### ISSUE TITLE
How to connect to official rabbitMQ Docker container

### ISSUE THREAD

**Issue by rajnemani** [squaremo/amqp.node#173]
Hello,

Can you please help with correct connection string to connect to official Docker container for abbitMQ (https://github.com/docker-library/rabbitmq)from a nodejs application using amqplib module. My container is running in Kitematic's Docker VM and my nodejs app is on the host running MAC OS X. I have passed in the -P flag when starting the container which set the port as 0.0.0.0:32768->5672/tcp. The connection string I am trying is of the form
amqp://guest:guest@hostName:32768/. I am unclear what should be the hostName here. I am also not sure if guest/guest are the right credentials. Any help is appreciated and thank you in advance.

Thanks

**Comment by michaelklishin** [comment]
This sounds like a Docker question more suitable for the Docker image users/maintainers (which is not official, by the way, since RabbitMQ team didn't produce it).

**Comment by rajnemani** [comment]
Hi Michael,
Thank you.  I have already asked the question there without results yet.  I will pursue further using other means,  You are correct that rabbitMQ did not produce this image.  Docker checks and maintains this image as one of the few official Docker images.  But this is a side note.  Thanks again for your time.

**Comment by squaremo** [comment]
@rajnemani I think you will want the kitematic VM's IP as the host. You _could_ also want to use `-p 5672:0.0.0.0:5672` (lowercase) to publish the container port on a specific host port, which would make things easier.

**Comment by rajnemani** [comment]
Hi Michael,
This worked fine.  After mapping the ports as above the following
connection string worked.

amqp://XXX.XXX.XXX.XXX

Note that amqp://XXX.XXX.XXX.XXX/ does not work (the extra forward slash at
the end. It has something to do with rabbitMQ vhost stuff, I am still
reading more on it ). I think this extra slash could have been my problem
all along.   I really appreciate your help.
Thanks again.
Raj

**Comment by squaremo** [comment]
> the extra forward slash at the end. It has something to do with rabbitMQ vhost stuff

Because of a quirk in the AMQP URL spec (which kind of follows from a quirk in the AMQP spec), the path part of the URL following a slash refers to the vhost, so a _trailing_ slash means "use a vhost of `''` (the empty string)"; while leaving the slash off means "use the default vhost". This runs slightly against intuition, because the default vhost is usually `'/'`.

If you find this hard to follow, you are not the only one! It is usually enough to remember that leaving off the slash will do the right thing for a default installation of RabbitMQ.

**Comment by rajnemani** [comment]
glad I am not the only one :)
Thanks again.
Raj

**Comment by squaremo** [comment]
I'm going to close this to take it off the radar, since you're on your way :)
